### ISSUE TITLE
CONTAINS("test", "") == false

### ISSUE THREAD

**Issue by Yogu** [arangodb/arangodb#5640]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.3

Mode:
- [x] Single-Server

Storage-Engine:
- [x] mmfiles

On this operating system:
- [x] Windows, version: 10

### this is an AQL-related issue:
[ ] I'm using graph features

I'm issuing AQL via:
- [x] web interface with this browser:     running on this OS:

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is:

    RETURN CONTAINS(test", "")

It returns `false`, but all languages I am aware of return true for a equivalent call.

**Comment by jsteemann** [comment]
@Yogu : it seems that not all languages do agree on this.
A quick survey reveals the following results:

PHP:
```
strpos('test', '');
PHP Warning:  strpos(): Empty needle in Command line code on line 1
bool(false)
strpos('test', 'x');
bool(false)
```
Python:
```
>>> print "test".find("")
0
>>> print "test".find("x")
-1
```
JavaScript:
```
"test".indexOf("")
0
"test".indexOf("x")
-1
```
C:
```
strstr("test", "")
pointer to first byte of "test"
strstr("test", "x")
null pointer
```
C++:
```
std::string("test").find("")
0
std::string("test").find("x")
std::string::end
```
Ruby:
```
irb(main):001:0> "test".include? ""
=> true
irb(main):002:0> "test".include? "x"
=> false
```

**Comment by Yogu** [comment]
Fair point, though I wouldn't say "PHP does it like this" is the best explanation in language design ;)

**Comment by Simran-B** [comment]
Shouldn't a function/method/operator be used that resembles a "contains" test (if available) rather than one that returns the position in the string if we compare it to AQL's `CONTAINS()` with returnIndex = false?

Modern JavaScript:

```
"test".includes("")
true
"test".includes("x")
false


// Bonus:
"test".includes()
false
"test".includes(null)
false
```

Python 2/3:

```
>>> "" in "test"
True
>>> "x" in "test"
False
```

**Comment by Yogu** [comment]
So we all agree that this should be changed? Don't know how backwards-compatibility concerns play a role here.

**Comment by Simran-B** [comment]
If queries match documents all of the sudden that did not match before, which seems totally possible with such a change, I would say this is indeed breaking backwards compatibility.

I wonder if this could also affect `null` values somehow. Or whether `null in "test"` should also return true...
